python/lsst/ap/verify/ingestion.py

```diff
@@ -454,7 +455,59 @@ def __init__(self, dataset, workspace, *args, **kwargs):
     def run(self):
         """Ingest the contents of a dataset into a Butler repository.
         """
-        pass
+        self._ensureRaws()
+
+    def _ensureRaws(self):
+        """Ensure that the repository in ``workspace`` has raws ingested.
+
+        After this method returns, this task's repository contains all science
+        data from this task's dataset. Butler operations on the repository are
+        not able to modify ``dataset`` in any way.
+
+        Raises
+        ------
+        RuntimeError
+            Raised if there are no files to ingest.
+        """
+        # I don't foresee multi-camera datasets, but guard against them just in case
+        rawCollectionFilter = re.compile(f"{self.dataset.instrument}/raw/\\w+")
+        rawCollections = [x for x in
+                          self.workspace.gen3WorkButler.registry.queryCollections(rawCollectionFilter)]
```

Comment:
The failure mode I was worried about was if there was an `cameraA/raw/all` collection, and that caused the task to skip ingesting data for `cameraB`.

As for replacing `all` with wildcards, that was because I'm not convinced the current naming scheme is final, and I don't want to have to chase naming conventions.

Having `Dataset.instrument` return an `Instrument` is an interesting idea, though. I'd prefer that if it's doable.